src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -68,14 +81,106 @@ public int DefaultBufferSize
         }
 
         /// <summary>
-        /// Determines whether null values of properties are ignored or whether they are written to the JSON.
+        /// Determines whether null values are ignored during serialization and deserialization.
+        /// The default value is false.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if this property is set after serialization or deserialization has occurred.
+        /// </exception>
+        public bool IgnoreNullValues
+        {
+            get
+            {
+                return _ignoreNullValues;
+            }
+            set
+            {
+                VerifyMutable();
+                _ignoreNullValues = value;
+            }
+        }
+
+        /// <summary>
+        /// Determines whether read-only properties are ignored during serialization and deserialization.
+        /// A property is read-only if it contains a public getter but not a public setter.
+        /// The default value is false.
         /// </summary>
-        public bool IgnoreNullPropertyValueOnWrite { get; set; }
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if this property is set after serialization or deserialization has occurred.
+        /// </exception>
+        public bool IgnoreReadOnlyProperties
+        {
+            get
+            {
+                return _ignoreReadOnlyProperties;
+            }
+            set
+            {
+                VerifyMutable();
+                _ignoreReadOnlyProperties = value;
+            }
+        }
 
         /// <summary>
-        /// Determines whether null values in the JSON are ignored or whether they are set on properties.
+        /// Gets or sets the maximum depth allowed when reading or writing JSON, with the default (i.e. 0) indicating a max depth of 64.
+        /// Reading past this depth will throw a <exception cref="JsonReaderException"/>.
         /// </summary>
-        public bool IgnoreNullPropertyValueOnRead { get; set; }
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if this property is set after serialization or deserialization has occurred.
+        /// </exception>
+        public int MaxDepth
+        {
+            get
+            {
+                return _maxDepth;
+            }
+            set
+            {
+                VerifyMutable();
+                _maxDepth = value;
+            }
+        }
+
+        /// <summary>
+        /// Defines how the <see cref="Utf8JsonReader"/> should handle comments when reading through the JSON.
```

Comment:
This is serializer, not reader. I get skip, but what does Allow mean in this context?

Reply:
The same as Skip... Unless we add callback support or other mechanism in the future.